cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/CombinedMetricsScanner.java

```diff
@@ -143,16 +147,18 @@ private Row getV2TableScannerNextRow() {
     try {
       return v2TableScanner.next();
     } catch (Exception e) {
-      if (e instanceof IOException) {
-        try {
-          if (!datasetFramework.hasInstance(v2MetricsTableDatasetId)) {
-            v2TableScanner.close();
-            v2TableScanner = null;
-            return null;
+      if (e instanceof DataSetException || e instanceof RuntimeException) {
+        if (e.getCause() != null && e.getCause() instanceof IOException) {
+          try {
+            if (!datasetFramework.hasInstance(v2MetricsTableDatasetId)) {
+              v2TableScanner.close();
+              v2TableScanner = null;
+              return null;
+            }
+          } catch (DatasetManagementException dme) {
+            LOG.error("Exception while checking dataset exists", dme);
```

Comment:
in this case, this exception is when we are checking if dataset instance exists.